dashboard/app/public_json_api.go

```diff
@@ -45,20 +46,31 @@ func getExtAPIDescrForBugPage(bugPage *uiBugPage) *publicAPIBugDescription {
 			}
 			return []string{bugPage.Bug.ExternalLink}
 		}(),
-		FixCommits: func() []fixCommit {
+		FixCommits: func() []vcsCommit {
 			if len(bugPage.Bug.Commits) == 0 {
 				return nil
 			}
-			var res []fixCommit
+			var res []vcsCommit
 			for _, commit := range bugPage.Bug.Commits {
-				res = append(res, fixCommit{
+				res = append(res, vcsCommit{
 					Title: commit.Title,
 					Hash:  commit.Hash,
 					Link:  commit.Link,
 				})
 			}
 			return res
 		}(),
+		CauseCommit: func() *vcsCommit {
+			if bugPage.BisectCause == nil || bugPage.BisectCause.Commit == nil {
+				return nil
+			}
+			commit := bugPage.BisectCause.Commit
+			return &vcsCommit{
```

Comment:
This is a good candidate for a helper function that create api commit from bisect commit, since it's duplicated twice now.

Reply:
Code is duplicated because I create vcsCommit from *dashapi.Commit and *uiCommit.
Types refactoring and deduplication seems to be the easiest way to fix it.
Do you see easier ways? (generics don't support struct fields access yet)

Reply:
@a-nogikh proposed to delete uiCommit and I actually like it. See https://github.com/google/syzkaller/pull/3988.

Reply:
done